Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -113,6 +113,31 @@ final class AsyncAwaitTests: MongoSwiftTestCase {
             }
         }
     }
+
+    func testConcurrentClientUsage() throws {
+        testAsync {
+            try await self.withTestNamespace { _, db, _ in
+                defer {
+                    db.syncDropOrFail()
+                }
+                try await withThrowingTaskGroup(of: Int.self) { group in
+                    for i in 1...20 {
```

Comment:
the choice of number of tasks and number of operations was somewhat arbitrary here, and I'm open to any suggestions on adjusting these if they seem unreasonable. this seemed to strike a reasonable balance between "high degree of concurrency" and "test that completes in a reasonable amount of time" (~3 seconds).